addons/l10n_id_efaktur/models/efaktur_document.py

```diff
@@ -0,0 +1,299 @@
+            })
+
+        self.message_post(
+            body=_("The e-Faktur report has been generated"),
+            attachments=[(attachment.name, attachment.raw)]
+        )
+
+    def _generate_efaktur_invoice(self, delimiter=','):
+        """Generate E-Faktur for customer invoice."""
+        # Invoice of Customer
+
+        output_head = '%s%s%s' % (
+            _csv_row(FK_HEAD_LIST, delimiter),
+            _csv_row(LT_HEAD_LIST, delimiter),
+            _csv_row(OF_HEAD_LIST, delimiter),
+        )
+
+        idr = self.env.ref('base.IDR')
+
+        for move in self.invoice_ids.filtered(lambda m: m.state == 'posted'):
+            eTax = move._prepare_etax()
+
+            commercial_partner = move.partner_id.commercial_partner_id
+            nik = str(commercial_partner.l10n_id_nik) if not commercial_partner.vat else ''
+
+            if move.l10n_id_replace_invoice_id:
+                number_ref = str(move.l10n_id_replace_invoice_id.name) + " replaced by " + str(move.name) + " " + nik
+            elif nik:
+                number_ref = str(move.name) + " " + nik
+            else:
+                number_ref = str(move.name)
+
+            street = ', '.join([x for x in (move.partner_id.street, move.partner_id.street2) if x])
+
+            invoice_npwp = ''
+            if commercial_partner.vat and len(commercial_partner.vat) >= 15:
+                invoice_npwp = commercial_partner.vat
+            elif commercial_partner.l10n_id_nik:
+                invoice_npwp = commercial_partner.l10n_id_nik
+            if not invoice_npwp:
+                action_error = {
+                    'view_mode': 'form',
+                    'res_model': 'res.partner',
+                    'type': 'ir.actions.act_window',
+                    'res_id': commercial_partner.id,
+                    'views': [[self.env.ref('base.view_partner_form').id, 'form']],
+                }
+                msg = _("Please make sure that you've input the appropriate NPWP or NIK for the following customer")
+                raise RedirectWarning(msg, action_error, _("Edit Customer Information"))
+            invoice_npwp = invoice_npwp.replace('.', '').replace('-', '')
+
+            etax_name = commercial_partner.name
+            if invoice_npwp[:15] == '000000000000000' and commercial_partner.l10n_id_nik:
+                etax_name = "%s#NIK#NAMA#%s" % (commercial_partner.l10n_id_nik, etax_name)
+
+            # Here all fields or columns based on eTax Invoice Third Party
+            eTax['KD_JENIS_TRANSAKSI'] = move.l10n_id_tax_number[0:2] or 0
+            eTax['FG_PENGGANTI'] = move.l10n_id_tax_number[2:3] or 0
+            eTax['NOMOR_FAKTUR'] = move.l10n_id_tax_number[3:] or 0
+            eTax['MASA_PAJAK'] = move.invoice_date.month
+            eTax['TAHUN_PAJAK'] = move.invoice_date.year
+            eTax['TANGGAL_FAKTUR'] = '{0}/{1}/{2}'.format(move.invoice_date.day, move.invoice_date.month, move.invoice_date.year)
```

Comment:
Could you please use a date formatting method instead ^^
```suggestion
            eTax['TANGGAL_FAKTUR'] = move.invoice_date.strftime(%d/%m/%Y')
```